src/app/users/login.ts

```diff
@@ -29,14 +25,11 @@ export const login = async ({
   logger,
   ip,
 }: {
-  phone: string
+  phone: PhoneNumber
```

Comment:
Why are you passing in `PhoneNumber`? Up to now we had discussed to do primitive -> domain type conversion within app layer.
Is this a special case or are we changing the pattern?

Reply:
the graphql layer provide a way to verify type, and we were already using it for phone, but we still kept the result as a string.

you'll see in the previous code that we were doing two (different (uh)) validations. a first one in the graphql layer with one regex, and another one in the domain with another regex.

https://github.com/GaloyMoney/galoy/pull/1382/files#diff-9c3159a2d8de10efb8a453dab443a53751163476970a8ac8eeac0a8ad7a42feeR22

I have converted some that just required one regex directly (or generally: no async/network call function) in the graphql layer so those types are already cast entering the domain layer

for types conversion that are more complex/require more complex validation I have left them in the domain layer.

Reply:
here is an example where the casting were done 3 times: https://github.com/GaloyMoney/galoy/pull/1382/commits/cbb8de8a945ff3acd26e30785abe256f482dcebe#diff-5e943a707e08464a698469ae3b768e75d563aa194102cf6ba398ae017abd2fe1L31

1. at the graphql layer
2. at the requestPhoneCodeWithCaptcha function
3. at the requestPhoneCode function